pandas/core/indexing.py

```diff
@@ -1936,42 +1935,31 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
         """
         pi = plane_indexer
 
-        ser = self.obj._ixs(loc, axis=1)
-
         # perform the equivalent of a setitem on the info axis
         # as we have a null slice or a slice with full bounds
         # which means essentially reassign to the columns of a
         # multi-dim object
         # GH#6149 (null slice), GH#10408 (full bounds)
         if com.is_null_slice(pi) or com.is_full_slice(pi, len(self.obj)):
-            ser = value
+            self.obj._iset_item(loc, value)
         elif (
             is_array_like(value)
-            and is_exact_shape_match(ser, value)
```

Comment:
looks like this is the only usage of is_exact_shape_match, so it could be removed

Reply:
looks like ATM we check for size matching, any reason that is dropped here?  (not clear why the existing check doesn't check check for shape match directly)

Reply:
The size checking is not dropped, I replaced it with

```
            and len(value.shape) > 0
            and self.obj.shape[0] == value.shape[0]
```

(but the diff shows it below this comment box)

The reason I replaced it with this more manual check, is because `ser` is no longer available